zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -0,0 +1,43 @@
+//! Contains code that interfaces with the zcash_primitives crate from
+//! librustzcash.
+
+use std::{
+    convert::{TryFrom, TryInto},
+    io,
+};
+
+use crate::{serialization::ZcashSerialize, transaction::Transaction};
+
+impl TryFrom<&Transaction> for zcash_primitives::transaction::Transaction {
+    type Error = io::Error;
+
+    /// Convert a Zebra transaction into a librustzcash one.
+    ///
+    /// # Panics
+    ///
+    /// If the transaction is not V5. (Currently there is no need for this
+    /// conversion for other versions.)
+    fn try_from(trans: &Transaction) -> Result<Self, Self::Error> {
+        let network_upgrade = match trans {
+            Transaction::V5 {
+                network_upgrade, ..
+            } => network_upgrade,
+            _ => panic!("Transaction must be V5"),
```

Comment:
Let's use an exhaustive match to turn this into a compilation error when we add `Transaction::V6`.
Otherwise, we might forget to update this code!

```suggestion
            Transaction::V5 {
                network_upgrade, ..
            } => network_upgrade,
            Transaction::V1 { .. } | Transaction::V2 { .. } |  Transaction::V3 { .. } |  Transaction::V4 { .. } => panic!("Zebra only uses librustzcash for V5 transactions"),
```